internal/cmd/group_utils.go

```diff
@@ -0,0 +1,52 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/tursodatabase/turso-cli/internal"
+	"github.com/tursodatabase/turso-cli/internal/prompt"
+	"github.com/tursodatabase/turso-cli/internal/turso"
+)
+
+func RetryOnSleepingGroup(client *turso.Client, group string, mainSpinnerText string, action func() (any, error)) (any, error) {
+	mainSpinner := prompt.Spinner(mainSpinnerText)
+	result, actionErr := action()
+	if actionErr == nil {
+		return result, nil
+	}
+
+	errMsg := actionErr.Error()
+	if strings.Contains(errMsg, "group_sleeping") ||
+		(strings.Contains(errMsg, "cannot create database on group") && strings.Contains(errMsg, "because it is archived")) {
```

Comment:
This will be reached, but I hate having to pass in the string and detect errors like this. I would much prefer to use `group_sleeping` from the error response `code` field.